### ISSUE TITLE
After log rotation fluentd tail plugin not working properly

### ISSUE THREAD

**Issue by moharana-subhashree** [fluent/fluentd#4306]
### Describe the bug

We observed that tail may stop processing the new log file after detecting log rotation.
and the fluentd worker process is getting killed, and then a new worker process starts.
whenever the log file rotated fluentd worker process gets killed after tailing the new log file .

### To Reproduce

1

### Expected behavior

fluentd should be able to read the new log file using tail.

### Your Environment

```markdown
- Fluentd version: 1.15.3 
- TD Agent version: 4.4.2 
- Operating system: 
- Kernel version:
```

### Your Configuration

```apache
<source>
      @type tail
      path /data0/podman/storage/overlay-containers/*/userdata/ctr.log
      pos_file /var/log/td-agent/podman.pos
      #read_from_head true
      refresh_interval 2s
      enable_stat_watcher false
      tag kubernetes.podman.*
      @log_level trace
      <parse>
        @type cri
      </parse>
    </source>
```

### Your Error Log

```shell
fluentd logs - 
`
{"log":{"message":"tailing paths: target = /data0/podman/storage/overlay-containers/40db0a0e893c0ad1ff0cc59c2cf4bb8b05e0a18b02b2e4910b2c197aedc33329/userdata/ctr.log,/data0/podman/storage/overlay-containers/6438ea18cea93a33a19cf6f3e55e2877767ffd6e7a749e7f2d6f91c7744d5c10/userdata/ctr.log | existing = /data0/podman/storage/overlay-containers/40db0a0e893c0ad1ff0cc59c2cf4bb8b05e0a18b02b2e4910b2c197aedc33329/userdata/ctr.log,/data0/podman/storage/overlay-containers/6438ea18cea93a33a19cf6f3e55e2877767ffd6e7a749e7f2d6f91c7744d5c10/userdata/ctr.log"},"extension":{"worker_id":0},"type":"log","level":"debug","timezone":"xx","system":"xx","systemid":"xx","host":"xx","time":"2023-09-21T20:11:00+0300"}
{"log":{"message":"detected rotation of /data0/podman/storage/overlay-containers/40db0a0e893c0ad1ff0cc59c2cf4bb8b05e0a18b02b2e4910b2c197aedc33329/userdata/ctr.log"},"extension":{"worker_id":0},"type":"log","level":"info","timezone":"Europe/Helsinki","system":"xx","systemid":"xx","host":"xx","time":"2023-09-21T20:11:01+0300"}
{"log":{"message":"following tail of /data0/podman/storage/overlay-containers/40db0a0e893c0ad1ff0cc59c2cf4bb8b05e0a18b02b2e4910b2c197aedc33329/userdata/ctr.log"},"extension":{"worker_id":0},"type":"log","level":"info","timezone":"Europe/Helsinki","system":"xx","systemid":"xx","host":"xx","time":"2023-09-21T20:11:01+0300"}
{"time":"2023-09-21T20:23:21+0300","level":"error","message":"Worker 0 finished unexpectedly with signal SIGKILL"}
`
after almost 10mins of log rotation, the fluentd worker process is also getting killed. and this is the pattern continuously happening after every log rotation.
```

### Additional context

when checked the pos file inside fluentd container ,
bash-4.4# cat /var/log/td-agent/podman.pos
`
/data0/podman/storage/overlay-containers/6438ea18cea93a33a19cf6f3e55e2877767ffd6e7a749e7f2d6f91c7744d5c10/userdata/ctr.log      0000000000000000        000000000451db16
/data0/podman/storage/overlay-containers/40db0a0e893c0ad1ff0cc59c2cf4bb8b05e0a18b02b2e4910b2c197aedc33329/userdata/ctr.log      00000000003d15a0        0000000000457028
`
`
bash-4.4# ls "/proc/$(pgrep -f 'td-agent.*--under-supervisor')/fd" -l | grep ctr.log
lr-x------. 1 root root 64 Sep 22 07:58 10 -> /data0/podman/storage/overlay-containers/40db0a0e893c0ad1ff0cc59c2cf4bb8b05e0a18b02b2e4910b2c197aedc33329/userdata/ctr.log
lr-x------. 1 root root 64 Sep 22 07:58 31 -> /data0/podman/storage/overlay-containers/6438ea18cea93a33a19cf6f3e55e2877767ffd6e7a749e7f2d6f91c7744d5c10/userdata/ctr.log

**Comment by daipom** [comment]
This is the same topic as #4304, right?

Since we can't judge whether this is a bug of Fluentd with this information,  could you please discuss this problem in #4304?
(Please confirm [contributing guidelines](https://github.com/fluent/fluentd/blob/d5df992d844af38233c4d3788de714d644ddb6c4/CONTRIBUTING.md))

**Comment by daipom** [comment]
Note: Known issues where `in_tai` may not detect log rotation have been resolved in Fluentd v1.16.2 and v1.16.3.
